cmd/juju/application/deployer/charm.go

```diff
@@ -484,3 +493,15 @@ func (c *charmStoreCharm) PrepareAndDeploy(ctx *cmd.Context, deployAPI DeployerA
 	c.origin = csOrigin
 	return c.deploy(ctx, deployAPI)
 }
+
+func isEmptyOrigin(origin commoncharm.Origin, source commoncharm.OriginSource) bool {
```

Comment:
Why not add an `IsEmpty` to the `Origin` type? Sounds like a useful helper to have

Reply:
I can do, but I'm not sure how valid it actually would be outside of this.

Reply:
Ok, guess if it's only used here we can leave it there.